packages/ra-ui-materialui/src/button/Button.tsx

```diff
@@ -36,11 +38,12 @@ export const Button = (props: ButtonProps) => {
         color = 'primary',
         size = 'small',
         to: locationDescriptor,
+        state = {},
         ...rest
     } = props;
     const translate = useTranslate();
     const translatedLabel = label ? translate(label, { _: label }) : undefined;
-    const linkParams = getLinkParams(locationDescriptor);
+    const linkParams = merge(getLinkParams(locationDescriptor), { state });
```

Comment:
I find it weird (dangerous?) to add this capability to `Button` while we only need it in `CreateButton`.

Reply:
What do you think @djhi, @slax57 about it? This is the behaviour I understood, isn't it?

Reply:
As our Button already supports and already uses the `getLinkParams`, I think it makes sense to do it here

Reply:
I think we should only add it if we need it.